internal/controller/dnsrecord_controller.go

```diff
@@ -89,56 +97,61 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	}
 
 	if !controllerutil.ContainsFinalizer(dnsRecord, DNSRecordFinalizer) {
+		dnsRecord.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(validationRequeueTime))
 		logger.Info("Adding Finalizer", "name", DNSRecordFinalizer)
 		controllerutil.AddFinalizer(dnsRecord, DNSRecordFinalizer)
 		err = r.Update(ctx, dnsRecord)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
-		return ctrl.Result{Requeue: true}, nil
+		return ctrl.Result{RequeueAfter: validationRequeueTime}, nil
 	}
 
 	var reason, message string
-	status := metav1.ConditionTrue
-	reason = "ProviderSuccess"
-	message = "Provider ensured the dns record"
 	err = dnsRecord.Validate()
 	if err != nil {
-		status = metav1.ConditionFalse
 		reason = "ValidationError"
 		message = fmt.Sprintf("validation of DNSRecord failed: %v", err)
-		setDNSRecordCondition(dnsRecord, string(conditions.ConditionTypeReady), status, reason, message)
-		return r.updateStatus(ctx, previous, dnsRecord)
+		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse, reason, message)
+		return r.updateStatus(ctx, previous, dnsRecord, "0s")
 	}
+
 	// Publish the record
-	err = r.publishRecord(ctx, dnsRecord)
+	requeueAfter, err := r.publishRecord(ctx, dnsRecord)
```

Comment:
ok so this makes sense 👍 so publish->applyChanges->returns requeue if changes applied

Reply:
yes, `publish` would make a decision on calling the `applyChanges` or going into the "not yet!" loop (got the record before we queued it). The `applyChanges` would either do changes and return validation requeue time or will report that no changes are needed and return a long timeout.